packages/vega-typings/types/spec/scale.d.ts

```diff
@@ -55,44 +56,64 @@ export type UnionSortField =
     };
 export type ScaleField = string | SignalRef;
 
-export type ScaleBins =
-  | (number | SignalRef)[]
-  | SignalRef
-  | {
-      step: number | SignalRef;
-      start?: number | SignalRef;
-      stop?: number | SignalRef;
-    };
+export interface ScaleBinParams {
+  /**
+   * The step size defining the bin interval width.
+   */
+  step: number | SignalRef;
+
+  /**
+   * The starting (lowest-valued) bin boundary.
+   *
+   * __Default value:__ The lowest value of the scale domain will be used.
+   */
+  start?: number | SignalRef;
 
-export type ScaleInterpolate =
+  /**
+   * The stopping (highest-valued) bin boundary.
+   *
+   * __Default value:__ The highest value of the scale domain will be used.
+   *
+   */
+  stop?: number | SignalRef;
+}
+
+export type ScaleBins = (number | SignalRef)[] | SignalRef | ScaleBinParams;
+
+export type ScaleInterpolateEnum =
   | 'rgb'
   | 'lab'
   | 'hcl'
   | 'hsl'
   | 'hsl-long'
   | 'hcl-long'
   | 'cubehelix'
-  | 'cubehelix-long'
-  | SignalRef
-  | {
-      type: 'rgb' | 'cubehelix' | 'cubehelix-long' | SignalRef;
-      gamma?: number | SignalRef;
-    };
-export interface DataRef {
+  | 'cubehelix-long';
+
+export interface ScaleInterpolateParams {
+  type: 'rgb' | 'cubehelix' | 'cubehelix-long' | SignalRef;
+  gamma?: number | SignalRef;
+}
+
+export type ScaleInterpolate = ScaleInterpolateEnum | SignalRef | ScaleInterpolateParams;
+
+export interface ScaleDataRef {
   data: string;
   field: ScaleField;
 }
-export type MultiDataRef =
-  | {
-      data: string;
-      fields: ScaleField[];
-    }
-  | {
-      fields: ((string | number | boolean)[] | DataRef | SignalRef)[];
-    };
-export type ScaleData =
-  | (DataRef & { sort?: SortField })
-  | (MultiDataRef & { sort?: UnionSortField });
+
+export interface ScaleMultiDataRef {
+  fields: ((string | number | boolean)[] | ScaleDataRef | SignalRef)[];
```

Comment:
Do we have a type for `string | number | boolean`?

Reply:
Not that I know of.

Reply:
Let's merge first, if you find one we can fix them later.